File: MyAlbum.WebSPA/Core/Models/Album.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace MyAlbum.Core.Models
{
    public class Album
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public ICollection<Photo> Photos { get; set; }

        public User Author { get; set; }

        public DateTime? CreatedDate { get; set; }

        public DateTime? ModifiedDate { get; set; }

        public Album()
        {
            Photos = new Collection<Photo>();
        }
    }
}

File: MyAlbum.WebSPA/ClientApp/src/app/models/album.ts
import { Photo } from './photo';
import { User } from "./user";

export interface AlbumQuery {
    categoryId: number;
    hasLocation: boolean;
    authorUserName: string;
}

export interface Album {
    id: number;
    name: string;
    createdDate: Date;
    modifiedDate: Date;
    author: User;
    mainPhoto: Photo;
    subPhotos: Photo[];
    photos: Photo[];
}

export interface SaveAlbum {
    id: number;
    name: string;
}